CSharpMath.Editor/MathKeyboard.cs

```diff
@@ -62,10 +63,10 @@ public class MathKeyboard<TFont, TGlyph> where TFont : IFont<TGlyph> {
         blinkTimer.Start();
         if (value != MathKeyboardCaretState.Hidden &&
            MathList.AtomAt(_insertionIndex) is Atoms.Placeholder placeholder) 
-          (placeholder.Nucleus, _caretState) =
+          (placeholder.Nucleus, placeholder.ForeColor, _caretState) =
             value == MathKeyboardCaretState.TemporarilyHidden
-            ? ("\u25A1", MathKeyboardCaretState.TemporarilyHidden)
-            : ("\u25A0", MathKeyboardCaretState.ShownThroughPlaceholder);
+            ? (LaTeXSettings.PlaceholderHidingNucleus, LaTeXSettings.PlaceholderHidingColor, MathKeyboardCaretState.TemporarilyHidden)
+            : (LaTeXSettings.PlaceholderFullShowNucleus, LaTeXSettings.PlaceholderFullShowColor, MathKeyboardCaretState.ShownThroughPlaceholder);
```

Comment:
Hidden/shown is about the caret. Using these terms to describe the placeholder is weird.